src/gui/URLEdit.h

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright (C) 2014 Felix Geyer <debfx@fobos.de>
+ *  Copyright (C) 2019 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef KEEPASSX_URLEDIT_H
+#define KEEPASSX_URLEDIT_H
+
+#include <QAction>
+#include <QLineEdit>
+#include <QPointer>
+
+// Use a separate class for the comparison function so tests can be run without
+// dependency to QLineEdit
+class ValidURL
+{
+public:
+    static bool validURL(const QString& urlField);
```

Comment:
This is an odd place to put this function. I recommend putting this in the Tools namespace.

Reply:
Thanks for the tip.